### PR TITLE
feat(pwm): allow specifying OutputDrive for PWM channels

### PR DESCRIPTION
**Pull request by igiona**
HI everyone,

in an application I'm working on, I need a way to change the driving strength of the GPIO Pins.
Since I did find a `safe` way to do so on my side, I went implementing it here.

For the `SequencePwm` I extended the `Config` with the driving strength, whereas in the `SimplePwm` I added the function to adapt it similarly to the other parameters `set_max_duty`. I omitted the getters because personally I think that it's a "YAGNI".
I did not introduce a `SimpleConfig` because from the code it looks like not to be there intentionally, and I did not want to brake the API.

Happy to change any of this based on your feedback.

Cheers!

### CODE REVIEW

File: embassy-nrf/src/pwm.rs [hunk 1]
```diff
@@ -815,6 +831,38 @@ impl<'d, T: Instance> SimplePwm<'d, T> {
         let max_duty = self.max_duty() as u32;
         clk / max_duty
     }
+
+    /// Sets the PWM-Channel0 output drive strength
+    #[inline(always)]
+    pub fn set_ch0_drive(&self, drive: OutputDrive) {
+        if let Some(pin) = &self.ch0 {
+            pin.conf().write(|w| w.drive().variant(convert_drive(drive)));
```
Comment (Dirbaio) [inline embassy-nrf/src/pwm.rs:839]: these should be .modify() so it leaves the other fields unchanged.

Comment (igiona) [reply embassy-nrf/src/pwm.rs]: Oh of course thanks, I wrongly assumed `write()` doing the job for me, I'm a newbie in embedded-rust world as you can tell 😄

**Review by Dirbaio — APPROVED** [review]
thanks!

**Comment by igiona** [comment]
@Dirbaio the CI failure looks quite unrelated... possibly a flaky test case?

**Comment by Dirbaio** [comment]
probably

**Pull request merged**